hazelcast/src/main/java/com/hazelcast/jet/impl/connector/SpecificPartitionsImapReaderPms.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.connector;
+
+import com.hazelcast.cluster.Address;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.function.BiFunctionEx;
+import com.hazelcast.internal.serialization.InternalSerializationService;
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.impl.connector.HazelcastReaders.LocalMapReaderFunction;
+import com.hazelcast.jet.impl.connector.ReadMapOrCacheP.LocalProcessorMetaSupplier;
+import com.hazelcast.jet.impl.connector.ReadMapOrCacheP.LocalProcessorSupplier;
+import com.hazelcast.jet.impl.connector.ReadMapOrCacheP.Reader;
+import com.hazelcast.security.permission.MapPermission;
+
+import javax.annotation.Nonnull;
+import java.security.Permission;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+import static com.hazelcast.jet.config.JobConfigArguments.KEY_REQUIRED_PARTITIONS;
+import static com.hazelcast.security.permission.ActionConstants.ACTION_CREATE;
+import static com.hazelcast.security.permission.ActionConstants.ACTION_READ;
+
+/**
+ * A meta-supplier that will compute required partitions to scan,
+ * if attribute partitioning strategy was applied.
+ */
+public abstract class SpecificPartitionsImapReaderPms<F extends CompletableFuture, B, R> extends LocalProcessorMetaSupplier<F, B, R> {
+    private transient int[] partitionsToScan;
+    private transient Map<Address, int[]> partitionAssignment;
+
+    private SpecificPartitionsImapReaderPms(
+            final String mapName,
+            final BiFunctionEx<HazelcastInstance, InternalSerializationService, Reader<F, B, R>> readerSupplier) {
+        super(readerSupplier);
+    }
+
+    @Override
+    public void init(@Nonnull Context context) throws Exception {
+        super.init(context);
+        Set<Integer> partitions = context.jobConfig().getArgument(KEY_REQUIRED_PARTITIONS);
```

Comment:
you cannot use that. In the future we may have a query that scans multiple IMaps (or the same IMap but multiple times eg. in `UNION ALL` query) and a different subset of partitions from each of them. This information must be per vertex.

Reply:
just created HZ-2693

Reply:
another case is for joins of filtered IMaps - each IMap may use different set of partitions

Reply:
Other than that, we should think how to conveniently forward per-vertex partitions in parental PR... Or in some follow-up PR.

Reply:
regarding your example: of course it better to scan N*N partitions instead of N*11, but it would be even better to scan N*1. so why implement an intermediate solution if we can implement the ultimate one (can we?)? Is it that much more complicated?

Reply:
>  if we can implement the ultimate one (can we?)?

We can, but it should be a purpose of parental PR or separate PR which change a way how do we passing a set of required partitions to scan in PlanExecutor. I like separation of concerns, so the scope of this PR is limited to map scan PMS.